### PR TITLE
fix($compile): correctly handle `null`/`undefined` href `attrs.$set()`

### PR DESCRIPTION
**Pull request by gkalpak**
Accidentally broken while backporting #14890.

Since #14890, `$$sanitizeUri()` can no longer handle `null`/`undefined` values.
In 1.7.x, there are no such calls.
In 1.6.x, there is still one such calls inside `Attributes.$set()`, so it needs to be adjusted accordingly.

### CODE REVIEW

File: src/ng/compile.js [hunk 1]
```diff
@@ -1723,7 +1723,7 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
         if ((nodeName === 'a' && (key === 'href' || key === 'xlinkHref')) ||
             (nodeName === 'img' && key === 'src')) {
           // sanitize a[href] and img[src] values
-          this[key] = value = $$sanitizeUri(value, key === 'src');
+          this[key] = value = (value == null) ? value : $$sanitizeUri(value, key === 'src');
```
Comment (mgol) [inline src/ng/compile.js]: Are those parens around `value == null` necessary? ;)

**Review by mgol — APPROVED** [review]
One remark but LGTM anyway.

File: src/ng/compile.js (hunk 1 above)
Comment (gkalpak) [reply src/ng/compile.js]: No, but look how much cleaner it looks with the parens 😛

Comment (mgol) [reply src/ng/compile.js]: De gustibus etc. but I'm not insisting. ;)

**Review by Narretz — APPROVED** [review]

**Pull request merged**